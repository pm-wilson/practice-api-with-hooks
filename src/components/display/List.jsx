import React from 'react';
import PropTypes from 'prop-types';
import { Link, BrowserRouter } from 'react-router-dom';

const List = ({ characters }) => {
  return (
    <div>
      <h3>Character List</h3>
      {characters.map((character) => {
        return (
          <div key={character._id}>
            <BrowserRouter><Link className="character-name" to={`/details/${character._id}`}>{character.name}</Link></BrowserRouter>
            <br />
          </div>
        );
      })}
    </div>
  );
};

List.propTypes = {
  characters: PropTypes.array,
};

export default List;
