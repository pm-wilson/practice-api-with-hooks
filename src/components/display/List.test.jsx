import React from 'react';
import { render } from '@testing-library/react';
import List from './List';

describe('List component', () => {
  const characters = [{ _id: 1, name: 'jim' }, { _id: 2, name: 'fred' }];

  it('renders List', () => {
    const { asFragment } = render(<List characters={characters}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
