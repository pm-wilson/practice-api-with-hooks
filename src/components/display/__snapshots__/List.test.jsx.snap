// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Character List
    </h3>
    <div>
      <a
        class="character-name"
        href="/details/1"
      >
        jim
      </a>
      <br />
    </div>
    <div>
      <a
        class="character-name"
        href="/details/2"
      >
        fred
      </a>
      <br />
    </div>
  </div>
</DocumentFragment>
`;
